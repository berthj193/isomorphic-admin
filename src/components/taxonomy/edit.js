import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import TaxonomyForm from './form';

import withReduxForm from '../formWrapper/withReduxForm';
import Button from '../uielements/button';

const Form = withReduxForm({
  form: 'editTaxonomy',
  enableReinitialize: true,
})(TaxonomyForm);

export default class EditTaxonomy extends PureComponent {
  static propTypes = {
    id: PropTypes.string,
    onDelete: PropTypes.func,
  };

  handleDelete = () => {
    const { id, onDelete } = this.props;
    return onDelete && onDelete(id);
  };

  render() {
    const { onDelete, ...otherProps } = this.props;
    return (
      <div className="edit-taxonomy">
        <h2>Edit Taxonomy</h2>
        <Form {...otherProps}>
          { onDelete && <Button className="delete-taxonomy" onClick={this.handleDelete}>Delete</Button>}
        </Form>
      </div>
    );
  }
}
