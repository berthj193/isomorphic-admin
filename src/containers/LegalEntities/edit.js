import React from 'react';
import PropTypes from 'prop-types';
import Form from './legalEntitiesForm';
import withStoresManager from '../../components/apolloProviders/withStoresManager';
import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';
import { updateLegalEntity, getList } from '../../graphql/legalEntities';

export class EditLegalEntity extends React.Component {
  static propTypes = {
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    error: PropTypes.string,
    mutate: PropTypes.func,
    legal_entities: PropTypes.arrayOf(PropTypes.object),
    match: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.string,
      }),
    }),
  };

  static defaultProps = {
    legal_entities: [],
  };

  legalEntityId = () => this.props.match.params.id;

  legalEntity = () =>
    this.props.legal_entities.find(({ id }) => id === this.legalEntityId()) || {};

  handleSubmit = values => {
    this.props.mutate({
      mutation: updateLegalEntity,
      variables: {
        ...values,
        id: this.props.id,
      },
      refetchQueries: [{ query: getList }],
      messages: ['Legal entity has been modified successfully'],
    });
  };

  render() {
    const { id, name, vat_number } = this.legalEntity();
    return (
      <React.Fragment>
        <LayoutContentWrapper style={{ minHeight: '100vh' }}>
          <LayoutContent>
            <h1>Legal entities</h1>
            <ButtonList
              buttons={[
                { link: `/dashboard/legal_entities/${id}`, title: 'Go back' },
              ]} />
            <Form
              onSubmit={this.handleSubmit}
              initialValues={{ name, vat_number }}
            />
          </LayoutContent>
        </LayoutContentWrapper>
      </React.Fragment>
    );
  }
}

export default withStoresManager(EditLegalEntity,
  ({ match: { params: { id } } }) => ({
    query: getList,
    variables: {
      id,
    },
  })
);
