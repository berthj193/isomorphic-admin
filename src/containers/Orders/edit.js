import React, { Component } from 'react';
import PropTypes from 'prop-types';

import OrderForm from './orderForm';

import withOrdersManager from '../../components/apolloProviders/withOrdersManager';

import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';

import { getDetails, updateOrders } from '../../graphql/orders';

export class EditOrder extends Component {
  static propTypes = {
    mutate: PropTypes.func,
    orders: PropTypes.arrayOf(PropTypes.object),
  };

  static defaultProps = {
    orders: [{}],
  };

  handleSubmit = values => {
    const { shipping_address_id, shipping_rate_id } = values;

    this.props.mutate({
      mutation: updateOrders,
      variables: {
        ...values,
        ids: [this.props.orders[0].id],
        shipping_address_id: parseInt(shipping_address_id, 0),
        shipping_rate_id: parseInt(shipping_rate_id, 0),
      },
      messages: ['Order modified successfully'],
    });
  };

  render() {
    const order = this.props.orders[0];
    const {
      customer: {
        email: guest_email,
        given_name: guest_given_name,
        family_name: guest_family_name,
        phone_number,
      } = {},
      shipping_address: {
        id: shipping_address_id,
      } = {},
      shipping_rate: {
        id: shipping_rate_id,
      } = {},
    } = order;

    const orders = {
      ...order,
      guest_email,
      guest_given_name,
      guest_family_name,
      phone_number,
      shipping_address_id,
      shipping_rate_id,
    };

    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Orders</h1>
          <ButtonList buttons={[{ link: `/dashboard/orders/${order.id}`, title: 'Go back' }]} />
          <OrderForm
            onSubmit={this.handleSubmit}
            initialValues={orders}
          />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withOrdersManager(EditOrder,
  ({ match: { params: { key } } }) => ({
    query: getDetails,
    variables: {
      id: key,
    },
  }),
);
