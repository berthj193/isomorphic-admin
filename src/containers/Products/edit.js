import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ProductForm from './productForm';
import PageLoading from '../PageLoading/PageLoading';

import withStoresManager from '../../components/apolloProviders/withStoresManager';
import storeQueries from '../../graphql/stores';
import productQueries from '../../graphql/products';

import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';
import Error from '../../components/error/error';

export class EditProduct extends Component {
  static propTypes = {
    mutate: PropTypes.func,
    stores: PropTypes.arrayOf(PropTypes.object),
    products: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
        default_name: PropTypes.string,
        default_description: PropTypes.string,
        default_master_sku: PropTypes.string.isRequired,
      }),
    ),
    match: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.string,
      }),
    }),
    error: PropTypes.string,
  };

  static defaultProps = {
    products: [],
  };

  getProduct() {
    return this.props.products.find(({ id }) => id === this.props.match.params.id);
  }

  handleSubmit = ({ default_name, default_description, default_master_sku }) =>
    this.props.mutate({
      mutation: productQueries.updateProduct(),
      variables: {
        id: this.props.match.params.id,
        default_name,
        default_description,
        default_master_sku,
      },
      refetchQueries: [
        { query: productQueries.getProducts() },
      ],
      messages: ['Product updated successfully'],
    });

  render() {
    if (!this.props.error && !this.getProduct()) {
      return (<PageLoading style={{ height: '50vh' }} />);
    }

    const { id } = this.props.products[0];

    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Product</h1>
          <ButtonList buttons={[{ link: `/dashboard/products/${id}`, title: 'Go back' }]} />
          <ProductForm
            onSubmit={this.handleSubmit}
            stores={this.props.stores}
            initialValues={this.getProduct()}
          />
          <Error error={this.props.error} />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withStoresManager(EditProduct, ([
  {
    query: storeQueries.storesAutocomplete,
  },
  {
    query: productQueries.getProducts(),
  },
]));
