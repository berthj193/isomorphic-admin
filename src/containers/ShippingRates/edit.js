import React, { Component } from 'react';
import PropTypes from 'prop-types';

import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';
import PageLoading from '../PageLoading/PageLoading';

import withStoresManager from '../../components/apolloProviders/withStoresManager';
import ShippingRateForm from './shippingRateForm';

import { getList, getDetails, updateShippingRate } from '../../graphql/shippingRates';

export class ShippingRateEdit extends Component {
  static propTypes = {
    mutate: PropTypes.func,
    shipping_rates: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      country: PropTypes.string,
      name: PropTypes.string,
      premium: PropTypes.bool,
      live: PropTypes.bool,
      created_at: PropTypes.string,
      warehouse: PropTypes.shape({
        id: PropTypes.string,
      }),
    })),
  };

  static defaultProps = {
    shipping_rates: [],
  };


  getShippingRate() {
    return this.props.shipping_rates[0];
  }

  handleSubmit = values => {
    this.props.mutate({
      mutation: updateShippingRate,
      variables: {
        ...values,
        price_pence: parseInt(values.price_pence, 0),
        min_items: parseInt(values.min_items, 0),
        max_items: parseInt(values.max_items, 0),
      },
      refetchQueries: [
        { query: getList },
      ],
      messages: ['Shipping rate updated successfully'],
    });
  };

  getInitialValues() {
    const { warehouse, live, premium, ...rest } = this.props.shipping_rates[0];
    return {
      ...rest,
      warehouse_id: warehouse && warehouse.id,
      live: live || false,
      premium: premium || false,
    };
  }

  render() {
    const shippingRate = this.getShippingRate();
    if (!shippingRate) {
      return (<PageLoading style={{ height: '50vh' }} />);
    }
    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Shipping rates</h1>
          <ButtonList buttons={[{ link: `/dashboard/shipping_rates/${shippingRate.id}`, title: 'Go back' }]} />
          <ShippingRateForm
            onSubmit={this.handleSubmit}
            initialValues={this.getInitialValues()}
          />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withStoresManager(ShippingRateEdit,
  props => ({
    query: getDetails,
    variables: {
      id: props.match.params.id,
    },
  })
);
