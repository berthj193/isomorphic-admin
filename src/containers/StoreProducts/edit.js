import React, { Component } from 'react';
import PropTypes from 'prop-types';

import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';

import withStoresManager from '../../components/apolloProviders/withStoresManager';
import Form from './storeProductForm';

import storeProductQueries from '../../graphql/storeProducts';
import _omit from 'lodash/omit';

export class StoreProductEdit extends Component {
  static propTypes = {
    mutate: PropTypes.func.isRequired,
    store_products: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      name: PropTypes.string,
      master_sku: PropTypes.string,
      product: PropTypes.shape({ name: PropTypes.string }),
      store: PropTypes.shape({ name: PropTypes.string }),
      store_product_variants: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string,
      })),
    })),
    match: PropTypes.shape({
      params: PropTypes.shape({
        store_key: PropTypes.string.isRequired,
      }),
    }),
  };

  static defaultProps = {
    store_products: [{}],
  };

  handleSubmit = values => {
    const { id } = this.props.store_products[0];
    this.props.mutate({
      mutation: storeProductQueries.updateStoreProduct,
      variables: {
        ...values,
        id: parseInt(id),
      },
      messages: ['Store product was updated successfully'],
    });
  };

  render() {
    const { id, ...storeProduct } = this.props.store_products[0];
    const initialValues = _omit(storeProduct, ['store', 'product', 'store_product_variants']);

    const { match: { params: { store_key } } } = this.props;
    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Store Products</h1>
          <ButtonList buttons={[{ link: `/dashboard/store_products/${store_key}/${id}`, title: 'Go back' }]} />
          <Form
            onSubmit={this.handleSubmit}
            initialValues={initialValues}
            edit
          />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withStoresManager(StoreProductEdit,
  ({ match: { params: { store_key } } }) => ({
    query: storeProductQueries.getDetails,
    variables: {
      store_key,
    },
  }),
);
