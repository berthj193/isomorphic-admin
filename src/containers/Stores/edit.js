import React from 'react';
import PropTypes from 'prop-types';
import Form from './storeForm';
import withStoresManager from '../../components/apolloProviders/withStoresManager';
import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import ButtonList from '../../components/uielements/buttonList';
import storeQueries from '../../graphql/stores';

export class EditStore extends React.Component {
  static propTypes = {
    stores: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
        key: PropTypes.string,
        warehouse: PropTypes.shape({ name: PropTypes.string }),
        company: PropTypes.shape({ name: PropTypes.string }),
        channel: PropTypes.shape({ name: PropTypes.string }),
        store_products: PropTypes.arrayOf(PropTypes.object),
        created_at: PropTypes.string,
      }),
    ),
    match: PropTypes.shape({
      params: PropTypes.shape({
        key: PropTypes.string,
      }),
    }),
    mutate: PropTypes.func,
  };

  static defaultProps = {
    stores: [],
  };

  getStoreKey = () =>
    this.props.match.params.key;

  getStore = () => {
    const storeKey = this.getStoreKey();
    return this.props.stores.find(({ key }) => key === storeKey);
  };

  handleSubmit = values => {
    const options = values.options.reduce((prev, curr) => ({
      ...prev,
      [curr.key]: curr.value,
    }), {});
    this.props.mutate({
      mutation: storeQueries.updateStore,
      variables: {
        ...values,
        options: JSON.stringify(options),
      },
      refetchQueries: [
        {
          query: storeQueries.getList,
          variables: {
            store_key: this.getStoreKey(),
          },
        },
      ],
      messages: ['Store modified successfully'],
    });
  };

  initialValues = () => {
    const {
      id, key, options = {},
      channel: { id: channel_id } = {},
      warehouse: { id: warehouse_id } = {},
      legal_entity: { id: legal_entity_id } = {},
    } = this.getStore() || {};
    return {
      id, key,
      channel_id,
      warehouse_id,
      legal_entity_id,
      options: Object.entries(options).reduce((prev, curr) => [
        ...prev,
        {
          key: curr[0],
          value: String(curr[1]),
        },
      ], []),
    };
  };

  render() {
    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Stores</h1>
          <ButtonList buttons={[{ link: `/dashboard/stores/${this.getStoreKey()}`, title: 'Go back' }]} />
          <Form
            onSubmit={this.handleSubmit}
            initialValues={this.initialValues()}
          />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withStoresManager(
  EditStore,
  ({ match: { params: { key } } }) =>
    ({
      query: storeQueries.getList,
      variables: {
        store_key: key,
      },
    })
);
