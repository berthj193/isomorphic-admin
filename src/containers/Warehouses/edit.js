import React from 'react';
import PropTypes from 'prop-types';
import Form from './warehouseForm';
import withStoresManager from '../../components/apolloProviders/withStoresManager';
import LayoutContentWrapper from '../../components/utility/layoutWrapper';
import LayoutContent from '../../components/utility/layoutContent';
import PageLoading from '../PageLoading/PageLoading';
import ButtonList from '../../components/uielements/buttonList';
import { updateWarehouse, getDetails } from '../../graphql/warehouses';

export class EditWarehouse extends React.Component {
  static propTypes = {
    id: PropTypes.string,
    error: PropTypes.string,
    mutate: PropTypes.func.isRequired,
    match: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.string,
      }),
    }),
    warehouses: PropTypes.arrayOf(PropTypes.object),
  };

  static defaultProps = {
    warehouses: [],
  };

  handleSubmit = values => {
    this.props.mutate({
      mutation: updateWarehouse,
      variables: {
        ...values,
        id: this.props.id,
      },
      messages: ['Warehouse modified successfully'],
    });
  };

  render() {
    if (!this.props.error && !this.props.warehouses[0]) {
      return (<PageLoading style={{ height: '50vh' }} />);
    }
    const { id, name, export_class } = this.props.warehouses[0];
    return (
      <LayoutContentWrapper style={{ minHeight: '100vh' }}>
        <LayoutContent>
          <h1>Warehouses</h1>
          <ButtonList buttons={[{ link: `/dashboard/warehouses/${id}`, title: 'Go back' }]} />
          <Form
            onSubmit={this.handleSubmit}
            initialValues={{ name, export_class }}
          />
        </LayoutContent>
      </LayoutContentWrapper>
    );
  }
}

export default withStoresManager(
  EditWarehouse,
  ({ match: { params: { id } } }) => ({
    query: getDetails,
    variables: {
      id,
    },
  }),
);
