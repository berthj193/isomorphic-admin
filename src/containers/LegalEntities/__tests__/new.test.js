import React from 'react';
import { EditLegalEntity } from '../edit';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<EditLegalEntity />', () => {
  const getLegalEntityDetails = () => ({
    id: 1,
    name: 'foo',
    vat_number: 'bar',
  });
  const getComponent = (props = {}) => shallow(
    <EditLegalEntity
      getLegalEntityDetails={getLegalEntityDetails}
      {...props}
    />
  );
  const mutate = jest.fn();

  beforeEach(() => {
    jest.resetAllMocks();

    mutate.mockReturnValue(Promise.resolve({ foo: 'bar' }));
  });

  it('should render properly', () => {
    expect(toJson(getComponent())).toMatchSnapshot();
  });

  it('should execute mutate function on submit', () => {
    const wrapper = getComponent({ mutate });
    wrapper.instance().handleSubmit({});
    expect(mutate).toBeCalled();
  });
});
