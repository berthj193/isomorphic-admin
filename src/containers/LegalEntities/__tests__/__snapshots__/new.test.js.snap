// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditLegalEntity /> should render properly 1`] = `
<React.Fragment>
  <LayoutWrapper
    style={
      Object {
        "minHeight": "100vh",
      }
    }
  >
    <LayoutContent>
      <h1>
        Legal entities
      </h1>
      <ButtonList
        buttons={
          Array [
            Object {
              "link": "/dashboard/legal_entities/undefined",
              "title": "Go back",
            },
          ]
        }
      />
      <ReduxFormWrapper
        initialValues={
          Object {
            "name": undefined,
            "vat_number": undefined,
          }
        }
        onSubmit={[Function]}
      />
    </LayoutContent>
  </LayoutWrapper>
</React.Fragment>
`;
